### PR TITLE
Add Dtmf Tone string comparison

### PR DESCRIPTION
**Pull request by wangrui-msft**
The Dtmf tone is casted to single char like "1", "#" etc. The collecton of dtmf tones are translated to string like "1234#*".

# Contributing to the Azure SDK

```
if (recognizeResult is CollectTonesResult collectToneRecognizedResult) 
{ 
     string toneResults = collectToneRecognizedResult.ConvertToString(); 
     if (toneResults.equals("56#")) 
    {
         // Do something;
    }

}
```

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/DtmfTone.cs [hunk 1]
```diff
@@ -13,5 +13,75 @@ namespace Azure.Communication.CallAutomation
     [JsonConverter(typeof(EquatableEnumJsonConverter<DtmfTone>))]
     public readonly partial struct DtmfTone
     {
+        /// <summary>
+        /// The Dtmf Tone in character format.
+        /// </summary>
+        public string ToChar()
```
Comment (cochi2) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/DtmfTone.cs:19]: I find confusing that the name is `ToChar` but it returns a `string`. 
Shouldn't it return a `char` instead?

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/DtmfTone.cs [hunk 2]
```diff
@@ -13,5 +13,75 @@ namespace Azure.Communication.CallAutomation
...
+        {
+            string originalValue = ToString();
+            return TransferToChar(originalValue);
+        }
+
+        /// <summary>
+        /// Convert the tone from string to char digit like "1", "2", "#".
+        /// </summary>
+        private static string TransferToChar(string value)
```
Comment (cochi2) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/DtmfTone.cs:28]: Probably `ConvertToChar` and returning a `char` value is better.

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/CollectTonesResult.cs [hunk 3]
```diff
@@ -14,5 +14,18 @@ public partial class CollectTonesResult : RecognizeResult
...
         [CodeGenMember("Tones")]
         public IReadOnlyList<DtmfTone> Tones { get; }
+
+        /// <summary>
+        /// Convert the collection of tones to a string like "12345#".
+        /// </summary>
+        public string ConvertToString()
+        {
+            string initial = "";
+            foreach (DtmfTone atone in Tones)
```
Comment (cochi2) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/CollectTonesResult.cs:24]: Would it work:
```
return string.join("", Tones.Select(aTone => aTone.ToChar());
```

**Review by jasonshave — COMMENTED** [review]
Can you provide a code sample showing the DtmfTone collection a developer would deal with and how this change would help them compare their expected input? Perhaps include this in the description of your PR.

I'm also wondering if you can provide a practical example where the English spelling of the DTMF tone would be helpful.

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/DtmfTone.cs (hunk 1 above)
Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/DtmfTone.cs]: changed to char

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/DtmfTone.cs (hunk 2 above)
Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/DtmfTone.cs]: changed to converttochar

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/CollectTonesResult.cs (hunk 3 above)
Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/Events/CollectTonesResult.cs]: changed to lambda.

**Review by atazimsft — APPROVED** [review]

**Pull request merged**